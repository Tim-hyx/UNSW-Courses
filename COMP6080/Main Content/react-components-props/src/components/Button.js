import React from 'react';

const Button = ({ background, click, title }) => {
  return (
  	<button
  	  style={{
  	    width: '100px',
  	    background: background
  	  }}
  	  onClick={click}
  	>
      {title ? title : 'Submit'}
    </button>
  );
}

export default Button; 