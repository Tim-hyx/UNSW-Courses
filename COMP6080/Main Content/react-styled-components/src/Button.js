import styled, { css, keyframes } from 'styled-components';

const removeDefaultButtonStyles = css`
  border: 0;
  padding: 0;
  font-family: inherit;
  font-weight: normal;
  background-color: unset;
  outline: none;
`;

export const BaseButton = styled.button`
  ${removeDefaultButtonStyles};
  cursor: pointer;
  padding: 12px;
  box-sizing: border-box;
  border-radius: 4px;
  font-weight: bold;
`;

const clickAnimation = keyframes`
  0% {
    transform: scale(1);
    border-radius: 4px;
    
  }

  50% {
    transform: scale(1.01);
  }

  75% {
    transform: scale(0.99);
  }

  100% {
    transform: scale(1);
  }
`;

function createHoverableButton(backgroundColor, hoverColor, inverted) {
  return styled(BaseButton)`
    color: ${inverted ? 'white' : 'black'};
    background-color: ${backgroundColor};

    &:hover {
      background-color: ${hoverColor};
      animation: ${props => props.animated ? css`${clickAnimation} 0.15s ease-in-out` : undefined};
    }
  `;
}

export const PrimaryButton = createHoverableButton(
  "#19aaee",
  "#33bbff",
  true,
);

export const SecondaryButton = createHoverableButton(
  "#d5d7d7",
  "#edf0f2",
  false,
);

export const ContrastButton = createHoverableButton(
  "#7d2ae8",
  "#8d39fa",
  true,
);
