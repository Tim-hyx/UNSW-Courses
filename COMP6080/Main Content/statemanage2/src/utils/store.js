import React from 'react'

export const StoreContext = React.createContext(null)

export default ({ children }) => {
  const [movies, setMovies] = React.useState([])
  const [tvShows, setTvShows] = React.useState([])

  const store = {
    movies: [movies, setMovies],
    tvShows: [tvShows, setTvShows],
  }

  return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>
}