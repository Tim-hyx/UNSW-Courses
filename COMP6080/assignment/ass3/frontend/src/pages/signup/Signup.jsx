import React, { useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import InputAdornment from "@material-ui/core/InputAdornment";
import EmailIcon from "@material-ui/icons/Email";
import LockIcon from "@material-ui/icons/Lock";
import PersonIcon from '@material-ui/icons/Person';
import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import { signin, alertError } from "../../redux/actions";

const useStyles = makeStyles(() => ({
    root: {
        // 64px is the height of navbar
        height: "calc(100vh - 64px)",
        justifyContent: "center",
        alignItems: "center",
    },
    form: {
        display: "flex",
        flexDirection: "column",
        maxWidth: 400,
        minWidth: 300,
    },
}));
const Signup = () => {
    const [email, setEmail] = useState();
    const [password, setPassword] = useState();
    const [passwordToVerify, setpasswordToVerify] = useState();
    const [name, setName] = useState();
    const classes = useStyles();
    const dispatch = useDispatch();
    const history = useHistory();
    return (
        <Grid container className={classes.root}>
            <form
                className={classes.form}
                onSubmit={(e) => {
                    e.preventDefault();

                    
                    if (password !== passwordToVerify) {
                        dispatch(alertError("Password mismatch"));
                        return false;
                    }
                    dispatch(signin(email, password, name));
                    history.push("/dashboard");
                    return true;
                }}
            >
                <TextField
                    label="Email"
                    type="email"
                    margin="normal"
                    required
                    InputProps={{
                        startAdornment: (
                            <InputAdornment position="start">
                                <EmailIcon />
                            </InputAdornment>
                        ),
                    }}
                    onChange={(event) => setEmail(event.target.value)}
                />
                <TextField
                    label="Password"
                    margin="normal"
                    type="password"
                    required
                    InputProps={{
                        startAdornment: (
                            <InputAdornment position="start">
                                <LockIcon />
                            </InputAdornment>
                        ),
                    }}
                    onChange={(event) => setPassword(event.target.value)}
                />
                <TextField
                    label="Type password again"
                    margin="normal"
                    type="password"
                    required
                    InputProps={{
                        startAdornment: (
                            <InputAdornment position="start">
                                <LockIcon />
                            </InputAdornment>
                        ),
                    }}
                    onChange={(event) => setpasswordToVerify(event.target.value)}
                />
                <TextField
                    label="Name"
                    margin="normal"
                    type="text"
                    required
                    InputProps={{
                        startAdornment: (
                            <InputAdornment position="start">
                                <PersonIcon />
                            </InputAdornment>
                        ),
                    }}
                    onChange={(event) => setName(event.target.value)}
                />
                <div style={{ height: 20 }} />
                <Button color="primary" variant="contained" type="submit">
                    Signup
                </Button>
            </form>
        </Grid>
    );
};

export default Signup;
