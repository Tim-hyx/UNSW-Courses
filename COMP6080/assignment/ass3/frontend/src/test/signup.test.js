import { mount } from "enzyme";
import { Provider } from "react-redux";
import Button from "@material-ui/core/Button";
import React from "react";
import TextField from "@material-ui/core/TextField";
import store from "../redux/stores";
import Signup from "../pages/signup/Signup";
// import Login from "../pages/login/Login";

// test signup button
describe('test signup page static html', () => {

    const wrapper = mount(
        <Provider store={store}>
            <Signup />
        </Provider>);
    
    it('Signup page should have a button to signup', () => {
       
        const signup = wrapper.children();
        const signupButton = signup.find(Button);
        expect(signupButton.text()).toBe("Signup");
        expect(signupButton).toMatchSnapshot();
    })

    it('Signup page has a textfield to input user email', () => {
       
        expect(wrapper.find(TextField).first().text()).toBe("Email *");
    });

    it('Signup page has a textfield to input user email', () => {
      
        expect(wrapper.find(TextField).first().text()).toBe("Email *");
    });

    it('Signup page has a textfield to input user password', () => {
     
        expect(wrapper.find(TextField).at(1).text()).toBe("Password *");
    });

    it('Signup page has a textfield to verify user password', () => {
      
        expect(wrapper.find(TextField).at(2).text()).toBe("Type password again *");
    });


    it('Signup page has a textfield to input user name', () => {
    
        expect(wrapper.find(TextField).last().text()).toBe("Name *");
    });
})
