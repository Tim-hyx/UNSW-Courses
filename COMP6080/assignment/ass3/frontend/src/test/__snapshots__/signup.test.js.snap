// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page Signup page should have a button to signup 1`] = `ReactWrapper {}`;

exports[`test signup page static html Signup page should have a button to signup 1`] = `ReactWrapper {}`;
